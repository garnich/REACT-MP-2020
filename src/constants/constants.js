export const LOAD_MOVIES = 'LOAD_MOVIES';
export const ACTIVE_MOVIE_ID = 'ACTIVE_MOVIE_ID';
export const ADD_NEW_MOVIE = 'ADD_NEW_MOVIE';
export const EDIT_MOVIE = 'EDIT_MOVIE';
export const DELETE_MOVIE = 'DELETE_MOVIE';
export const MOVIE_SHOW_DETAILS = 'MOVIE_SHOW_DETAILS';
export const MOVIE_HIDE_DETAILS = 'MOVIE_HIDE_DETAILS';
export const SORT_MOVIES = 'SORT_MOVIES';
export const FILTER_MOVIES = 'FILTER_MOVIES';
