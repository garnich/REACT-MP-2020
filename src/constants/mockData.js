const movies = [
    {
        id: 1,
        title: 'Movie Title 1',
        release_date: '01-01-2001',
        genres: ['Adventure', 'Action'],
        poster_path: 'https://posterpathmovie1.com',
        overview: 'Lorem ipsum 1',
        runtime: 110,
    },
    {
        id: 2,
        title: 'Movie Title 2',
        release_date: '02-02-2002',
        genres: ['War', 'History'],
        poster_path: 'https://posterpathmovie2.com',
        overview: 'Lorem ipsum 2',
        runtime: 120,
    },
    {
        id: 3,
        title: 'Movie Title 3',
        release_date: '03-03-2003',
        genres: ['Drama', 'Action'],
        poster_path: 'https://posterpathmovie3.com',
        overview: 'Lorem ipsum 3',
        runtime: 130,
    }
];

export { movies }