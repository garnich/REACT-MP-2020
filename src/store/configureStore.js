import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import reducer from '../reducer/reducer';

export default (initialState) => {
    const store = createStore(reducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
};
