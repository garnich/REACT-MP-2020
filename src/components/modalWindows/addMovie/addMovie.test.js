import React from 'react';
import Enzyme, { render } from 'enzyme';
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import AddMovie from './addMovie';
import { movies } from './../../../constants/mockData';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect,
}));

const handleCloseSpy = jest.fn();
const handleClose = handleCloseSpy;
const initialState = {
    movies: [],
    activeId: null,
    sortedAndFilteredMovies: [],
    sorter: null,
    filter: 'all',
};

const store = configureStore([])(initialState);
    store.dispatch = jest.fn();
    store.subscribe = jest.fn();
    store.getState = jest.fn();

describe('Test <AddMovie/> component', () => {

    it('<AddMovie> test snapshot should render editMovie modal component', () => {
        const setStore = {
            ...store,
            movies,
            activeId: 1,
            sortedAndFilteredMovies: movies,
            sorter: null,
            filter: 'all',
        };
        const wrapper = render(
            <Provider store={setStore}>
                <AddMovie 
                    handleClose={handleClose}
                    editMovie={true}
                />
            </Provider>
        )
    
        expect(wrapper).toMatchSnapshot()
    });

    it('<AddMovie> test snapshot should render addMovie modal component', () => {
        const setStore = {
            ...store,
            movies,
            activeId: null,
            sortedAndFilteredMovies: movies,
            sorter: null,
            filter: 'all',
        };
        const wrapper = render(
            <Provider store={setStore}>
                <AddMovie 
                    handleClose={handleClose}
                    editMovie = {false}
                />
            </Provider>
        )
    
        expect(wrapper).toMatchSnapshot()
    });
});