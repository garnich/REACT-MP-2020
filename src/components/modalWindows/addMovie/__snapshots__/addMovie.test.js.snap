// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddMovie/> component <AddMovie> test snapshot should render addMovie modal component 1`] = `
<form
  class="col-12"
  role="form"
>
  <h2
    class="title"
  >
    Add movie
  </h2>
  <div
    class="form-group"
  >
    <label
      for="title"
    >
      Title
    </label>
    <input
      class="form-control "
      id="title"
      name="title"
      placeholder="Movie title here"
      type="text"
      value=""
    />
    <label
      for="date"
    >
      Relase date
    </label>
    <input
      class="form-control "
      id="date"
      name="release_date"
      placeholder="Select date"
      type="date"
      value=""
    />
    <label
      for="url"
    >
      Movie poster url
    </label>
    <input
      class="form-control "
      id="url"
      name="poster_path"
      placeholder="Movie URL here"
      type="url"
      value=""
    />
    <label
      for="genre"
    >
      Genre
    </label>
    <input
      class="form-control "
      id="genreInput"
      name="genres"
      placeholder="Select genre"
      type="text"
      value=""
    />
    <label
      for="overview"
    >
      Overview
    </label>
    <input
      class="form-control "
      id="overview"
      name="overview"
      placeholder="Overview here"
      type="text"
      value=""
    />
    <label
      for="runtime"
    >
      Runtime
    </label>
    <input
      class="form-control "
      id="runtime"
      name="runtime"
      placeholder="Runtime here"
      type="number"
    />
  </div>
  <div
    class="d-flex justify-content-end"
  >
    <button
      class="btn reset"
      type="reset"
    >
      Reset
    </button>
    <button
      class="btn save"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`Test <AddMovie/> component <AddMovie> test snapshot should render editMovie modal component 1`] = `
<form
  class="col-12"
  role="form"
>
  <h2
    class="title"
  >
    Edit movie
  </h2>
  <div
    class="form-group"
  >
    <label
      for="id"
    >
      Movie ID
    </label>
    <p
      id="id"
      name="id"
    />
    <label
      for="title"
    >
      Title
    </label>
    <input
      class="form-control "
      id="title"
      name="title"
      placeholder="Movie title here"
      type="text"
      value=""
    />
    <label
      for="date"
    >
      Relase date
    </label>
    <input
      class="form-control "
      id="date"
      name="release_date"
      placeholder="Select date"
      type="date"
      value=""
    />
    <label
      for="url"
    >
      Movie poster url
    </label>
    <input
      class="form-control "
      id="url"
      name="poster_path"
      placeholder="Movie URL here"
      type="url"
      value=""
    />
    <label
      for="genre"
    >
      Genre
    </label>
    <input
      class="form-control "
      id="genreInput"
      list="genres"
      name="genres"
      placeholder="Select genre"
      type="text"
      value=""
    />
    <datalist
      id="genres"
    />
    <label
      for="overview"
    >
      Overview
    </label>
    <input
      class="form-control "
      id="overview"
      name="overview"
      placeholder="Overview here"
      type="text"
      value=""
    />
    <label
      for="runtime"
    >
      Runtime
    </label>
    <input
      class="form-control "
      id="runtime"
      name="runtime"
      placeholder="Runtime here"
      type="number"
    />
  </div>
  <div
    class="d-flex justify-content-end"
  >
    <button
      class="btn reset"
      type="reset"
    >
      Reset
    </button>
    <button
      class="btn submit"
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
